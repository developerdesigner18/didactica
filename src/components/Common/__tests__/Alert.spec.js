import { createLocalVue, mount } from "@vue/test-utils";
import { waitRAF } from "@/../tests/utils.js";

import { BAlert } from "bootstrap-vue";
import LxpAlert from "../Alert.vue";

// create an extended `Vue` constructor
const localVue = createLocalVue();

localVue.component("lxp-alert", LxpAlert);
localVue.component("b-alert", BAlert);

describe("Alert.vue", () => {
  it("render when created ", async () => {
    let wrapper = await getWrapper();
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  });

  describe("Props", () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });
    it("should render the title based on prop", async () => {
      wrapper = await getWrapper();
      const titleText = wrapper.find(".lxp-alert__content--right h5");
      expect(titleText.text()).toBe("Welcome New");
    });
    it("should render the subtitle based on prop", async () => {
      wrapper = await getWrapper();
      const titleText = wrapper.find(".lxp-alert__content--right p");
      expect(titleText.text()).toBe("this is a sample text");
    });

    it("should render the image", async () => {
      wrapper = await getWrapper();
      await wrapper.setProps({
        imgPath:
          "https://files.lxp.academy.who.int/didactica/assets/images/course-card/course-illustration.svg"
      });
      await waitRAF();
      const imgPath = wrapper.find(".lxp-alert__content img");
      expect(imgPath.exists()).toBe(true);
    });
    it("should update based on the variant", async () => {
      wrapper = await getWrapper();
      await wrapper.setProps({
        variant: "secondary1"
      });
      await waitRAF();
      const variant = wrapper.find(".alert-secondary1");
      expect(variant.exists()).toBe(true);
    });
  });
});

function getWrapper() {
  let wrapper = mount(LxpAlert, {
    propsData: {
      title: "Welcome New",
      subTitle: "this is a sample text"
    },
    localVue
  });

  return wrapper;
}
