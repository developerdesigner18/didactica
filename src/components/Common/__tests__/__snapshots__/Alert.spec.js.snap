// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert.vue render when created  1`] = `
<div
  class="lxp-alert"
>
  <transition-stub
    css="true"
    enteractiveclass=""
    enterclass=""
    entertoclass="show"
    leaveactiveclass=""
    leaveclass="show"
    leavetoclass=""
    name=""
  >
    <div
      aria-atomic="true"
      aria-live="polite"
      class="alert alert-dismissible alert-primary"
      role="alert"
    >
      <button
        aria-label="Close"
        class="close"
        type="button"
      >
        <img
          alt="close"
          loading="lazy"
          src="https://files.lxp.academy.who.int/didactica/assets/images/modal/close.svg"
        />
      </button>
       
      <div
        class="lxp-alert__content"
      >
        <!---->
         
        <div
          class="lxp-alert__content--right"
        >
          <h5>
            Welcome New
          </h5>
           
          <p>
            this is a sample text
          </p>
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;
